src/context/librariesContext.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { createContext, Dispatch, useEffect, useState } from 'react';
+import { LibraryData } from '../components/RegistryAdmin';
+
+export type LibrariesContextValues = {
+  librariesInContext: LibraryData[];
+  setLibrariesInContext: Dispatch<React.SetStateAction<LibraryData[]>>;
+  setUpdatedLibrary: Dispatch<React.SetStateAction<LibraryData | undefined>>;
+};
+
+type LibrariesProviderProps = {
+  children: React.ReactNode;
+};
+
+export const LibrariesContext = createContext<LibrariesContextValues | null>(
+  null
+);
+
+export const LibrariesProvider = ({ children }: LibrariesProviderProps) => {
+  const [updatedLibrary, setUpdatedLibrary] = useState<
+    LibraryData | undefined
+  >();
+  const [librariesInContext, setLibrariesInContext] = useState<LibraryData[]>(
+    []
+  );
+
+  useEffect(() => {
+    if (updatedLibrary) {
+      // Find the index of the library with the same uuid as the one
+      // that was recently updated.
+      const indexOfElementToReplace = librariesInContext.findIndex(
+        (library) => library.uuid === updatedLibrary.uuid
+      );
+
+      // Replace the library at that index with the updatedLibrary.
+      const updatedLibrariesInContext = librariesInContext.slice();
```

Comment:
Interesting! I see what you're doing here. I'm not sure it's necessary to create a shallow copy of the libraries before updating the new library value. You could probably just splice() and then `setLibrariesInContext(librariesInContext)`, but I don't think this is bad at all! Very functional programming-esque.

Reply:
I was attempting to not mutate the state 🤷‍♀️

Reply:
Yeah I actually like it! Going to use in the future.

Reply:
Great!